play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/ArscResource.java

```diff
@@ -0,0 +1,84 @@
+package com.google.wear.watchface.dfx.memory;
+
+import java.nio.file.Path;
+
+/**
+ * Represents a resource, as defined in the resources.arsc file within an Android package.
+ */
+public class ArscResource {
+    // Resource type, for example "raw", "asset", "drawable" etc.
+    private final String resourceType;
+
+    // Resource name, for example "watchface" for res/raw/watchface.xml.
+    private final String resourceName;
+
+    // File extension of the resource, for example "xml" for res/raw/watchface.xml
+    private final String extension;
+
+    // Path in the package. This is the obfuscated path to the actual data, where obfuscation has
+    // been used, for example "res/raw/watchface.xml" may point to something like "res/li.xml".
+    private final Path filePath;
+
+    // The resource data itself.
+    private final byte[] data;
+
+    public ArscResource(
+            String resourceType,
+            String resourceName,
+            String extension,
+            Path filePath,
+            byte[] data
+    ) {
+        this.resourceType = resourceType;
+        this.resourceName = resourceName;
+        this.extension = extension;
+        this.filePath = filePath;
+        this.data = data;
+    }
+
+    public String getResourceType() {
+        return resourceType;
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public String getExtension() {
+        return extension;
+    }
+
+    public Path getFilePath() {
+        return filePath;
+    }
+
+    public byte[] getData() {
+        return data;
+    }
+
+    public Boolean isWatchFaceXml() {
```

Comment:
Is this actually the spec?

Reading https://developer.android.com/training/wearables/wff/setup#declare-shape-support

I wonder if it's actually

a) res/raw/watchface.xml - by default
b) res/raw/X.xml - where X is one of the entries in res/xml/watch_face_shapes.xml

Reply:
You're correct. This should definitely be improved: I've encountered issues where I've (accidentally) left other XML files in the raw directory and consequently failed validation as they have been misinterpreted as watch face XML files.

For now, I'll comment that this needs improving and open an issue: https://github.com/google/watchface/issues/23